librz/analysis/var_global.c

```diff
@@ -0,0 +1,419 @@
+	rz_return_val_if_fail(name && addr, NULL);
+	RzAnalysisVarGlobal *glob = RZ_NEW0(RzAnalysisVarGlobal);
+	if (!glob) {
+		return NULL;
+	}
+	glob->name = strdup(name);
+	glob->addr = addr;
+	if (comment) {
+		glob->comment = strdup(comment);
+	}
+	return glob;
+}
+
+/**
+ * \brief Add the global variable into hashtable
+ * 
+ * \param analysis RzAnalysis
+ * \param global_var Global variable instance
+ * \return true if succeed
+ */
+RZ_API RZ_OWN bool rz_analysis_var_global_add(RzAnalysis *analysis, RzAnalysisVarGlobal *global_var) {
+	rz_return_val_if_fail(analysis && global_var, false);
+	if (rz_analysis_var_global_get_byaddr(analysis, global_var->addr)) {
+		eprintf("Global variable at 0x%" PFMT64x " is already exist!\n", global_var->addr);
+		return false;
+	}
+	return ht_pp_insert(analysis->ht_global_var, global_var->name, global_var);
+}
+
+/**
+ * \brief Free the global variable instance
+ * 
+ * \param glob Global variable instance
+ * \return void
+ */
+RZ_API void rz_analysis_var_global_free(RzAnalysisVarGlobal *glob) {
+	if (!glob) {
+		return;
+	}
+
+	RZ_FREE(glob->name);
+	RZ_FREE(glob->comment);
+	rz_type_free(glob->type);
+	rz_analysis_var_global_clear_accesses(glob);
+	rz_vector_fini(&glob->constraints);
+	RZ_FREE(glob);
+}
+
+/**
+ * \brief Delete and free the global variable by its name
+ * 
+ * \param analysis RzAnalysis
+ * \param name Global Variable name
+ * \return true if succeed
+ */
+RZ_API bool rz_analysis_var_global_delete_byname(RzAnalysis *analysis, const char *name) {
+	rz_return_val_if_fail(analysis && name, false);
+	RzAnalysisVarGlobal *glob = rz_analysis_var_global_get_byname(analysis, name);
+	if (!glob) {
+		eprintf("No such global variable!\n");
+		return true;
+	}
+	rz_analysis_var_global_free(glob);
+	return ht_pp_delete(analysis->ht_global_var, name);
+}
+
+/**
+ * \brief Same as rz_analysis_var_global_delete_byname but by its address
+ * 
+ * \param analysis RzAnalysis
+ * \param addr Global Variable address
+ * \return true if succeed
+ */
+RZ_API bool rz_analysis_var_global_delete_byaddr(RzAnalysis *analysis, ut64 addr) {
+	rz_return_val_if_fail(analysis, false);
+	RzAnalysisVarGlobal *glob = rz_analysis_var_global_get_byaddr(analysis, addr);
+	if (!glob) {
+		eprintf("No such global variable!\n");
+		return true;
+	}
+	bool deleted = ht_pp_delete(analysis->ht_global_var, glob->name);
+	rz_analysis_var_global_free(glob);
+	return deleted ? true : false;
+}
+
+/**
+ * \brief Get the instance of global variable by its name
+ * 
+ * \param analysis RzAnalysis
+ * \param name Global variable name
+ * \return RzAnalysisVarGlobal *
+ */
+RZ_API RZ_BORROW RzAnalysisVarGlobal *rz_analysis_var_global_get_byname(RzAnalysis *analysis, const char *name) {
+	rz_return_val_if_fail(analysis && name, NULL);
+	RzAnalysisVarGlobal *glob;
+	bool found;
+	glob = ht_pp_find(analysis->ht_global_var, name, &found);
+	return found ? glob : NULL;
```

Comment:
```suggestion
	return (RzAnalysisVarGlobal *)ht_pp_find(analysis->ht_global_var, name, NULL);
```
No need to check for that boolean, since it will return a NULL when fail, and even if you find a value and is NULL, then the returned pointer is still NULL.